### PR TITLE
add project point to plane

### PR DESCRIPTION
**Pull request by chuongmep**
### What type of PR is this? (check all applicable)

- [ ] ♻️ Refactor
- [x] ✨ Feature
- [ ] 🐛 Bug Fix
- [ ] 👷 Optimization
- [ ] 📝 Documentation Update
- [ ] 🔖 Release
- [ ] 🚩 Other

### Description

Add function project a point to plane

### Added tests?

- [x] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📓 docs
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/GShark/Geometry/Point3.cs [hunk 1]
```diff
@@ -527,6 +527,20 @@ public double DistanceTo(Point3 other)
             }
             return d;
         }
+        
+        /// <summary>
+        /// Project a point onto a plane.
```
Comment (sonomirco) [inline src/GShark/Geometry/Point3.cs:532]: Projects

File: src/GShark/Geometry/Point3.cs [hunk 2]
```diff
@@ -527,6 +527,20 @@ public double DistanceTo(Point3 other)
...
+        /// </summary>
+        /// <param name="point">point</param>
+        /// <param name="plane">plane to project onto</param>
+        /// <returns name="the point projected to plane"></returns>
+        public static Point3 ProjectTo(Point3 point, Plane plane)
```
Comment (sonomirco) [inline src/GShark/Geometry/Point3.cs:537]: Could we call it ProjectToPlan and pass only a plane as input?

**Comment by sonomirco** [comment]
Thanks, @chuongmep, for the collaboration!! 👍

File: src/GShark/Geometry/Point3.cs (hunk 2 above)
Comment (chuongmep) [reply src/GShark/Geometry/Point3.cs]: Thanks for your review, updated now.

File: src/GShark/Geometry/Point3.cs [hunk 3]
```diff
@@ -527,6 +527,19 @@ public double DistanceTo(Point3 other)
             }
             return d;
         }
+        
+        /// <summary>
+        /// Projects a point onto a plane.
+        /// </summary>
+        /// <param name="plane">plane to project onto</param>
```
Comment (sonomirco) [inline src/GShark/Geometry/Point3.cs:534]: Plane to project onto.

File: src/GShark/Geometry/Point3.cs [hunk 4]
```diff
@@ -527,6 +527,19 @@ public double DistanceTo(Point3 other)
...
+        /// <returns name="the point projected to plane"></returns>
```
Comment (sonomirco) [inline src/GShark/Geometry/Point3.cs:535]: The point projected to plane.

**Review by sonomirco — APPROVED** [review]

**Pull request merged**